arbitrum-docs/arbos/l1-to-l2-messaging.mdx

```diff
@@ -196,4 +320,77 @@ modifier onlyFromMyL1Contract() override {
+| **To what degree is state mirrored between L1 and L2 nodes?**              | • No direct state mirroring<br />• L1 nodes maintain the root hash of L2 state as part of optimistic rollup mechanism                               | • No direct state mirroring<br />• L2 state changes are represented in L1 via root hash updates                                                               |
+| **Where is the contract's state stored?**                                  | • On L1 nodes<br />• Each Ethereum node stores the full state                                                                                       | • On L2 nodes<br />• Each Arbitrum node stores the full state                                                                                                 |
+| **How does the contract deployment impact L1's physical state?**           | • Increases storage size<br />• Changes account balances as per the contract deployment cost                                                        | • Indirectly, by generating a commitment on L1<br />• The contract's bytecode, deployed on L2, results in a Merkle proof on L1                                |
+| **How does the contract deployment impact L2's physical state?**           | • Indirectly, if the L1 contract interacts with L2<br />• Changes in L1 contracts could be relayed to L2 via Retryable Tickets                      | • Increases storage size<br />• Changes account balances as per the contract deployment cost                                                                  |
+| **When is gas paid?**                                                      | • During contract deployment<br />• During contract execution                                                                                       | • During contract deployment<br />• During contract execution                                                                                                 |
+| **Where is gas paid?**                                                     | • On Ethereum L1<br />• Paid to Ethereum validators                                                                                                 | • On Arbitrum L2<br />• Paid to Arbitrum validators                                                                                                           |
+| **Who pays gas?**                                                          | • Contract deployer<br />• Transaction initiator                                                                                                    | • Contract deployer<br />• Transaction initiator                                                                                                              |
```

Comment:
Info in the table from here and above is good and useful, but may be too general / off-topic for the L1-to-L2 message page; maybe it lives elsewhere?